File: apps/zap-downloader-node/src/commands/info.ts
import chalk from 'chalk';
import { fetchZapVersions } from '../parser';
import { formatBytes } from '../downloader';
import { Arguments } from 'yargs';
import { getProxyUrl } from '../proxy';

export const command = 'info';
export const describe = 'Show information about a specific addon';

export const builder = (yargs: any) => {
  return yargs.option('addon', {
    alias: 'a',
    description: 'Addon ID',
    type: 'string',
    demandOption: true,
  });
};

export const handler = async (argv: Arguments & {
  addon: string;
  proxy?: string;
}) => {
  const addonId = argv.addon;
  const proxy = argv.proxy || getProxyUrl();

  console.log('Fetching ZAP versions...');
  const zapVersions = await fetchZapVersions(proxy);

  const addon = zapVersions.addons.find(a => a.id === addonId);
  if (!addon) {
    console.error(chalk.red(`Addon not found: ${addonId}`));
    process.exit(1);
  }

  console.log(`\n=== ${addon.name} ===`);
  console.log(`ID: ${addon.id}`);
  console.log(`Version: ${addon.version}`);
  console.log(`Status: ${addon.status}`);
  console.log(`Author: ${addon.author}`);
  console.log(`Description: ${addon.description}`);
  console.log(`File: ${addon.file}`);
  console.log(`Size: ${formatBytes(addon.size)}`);
  console.log(`Hash: ${addon.hash}`);
  console.log(`Date: ${addon.date}`);
  console.log(`Min ZAP Version: ${addon.notBeforeVersion}`);
  if (addon.dependencies && addon.dependencies.length > 0) {
    console.log('Dependencies:');
    for (const dep of addon.dependencies) {
      console.log(`  - ${dep.id} ${dep.version}`);
    }
  }
  console.log(`URL: ${addon.url}`);
};

File: apps/zap-downloader/zap_downloader/commands/info.py
import typer
import asyncio
from rich.console import Console

from ..parser import fetch_zap_versions

console = Console()


def info(
    addon_id: str = typer.Argument(..., help="Addon ID to get info about"),
):
    """Get detailed information about a specific addon."""
    asyncio.run(_get_info(addon_id))


async def _get_info(addon_id: str):
    from ..downloader import format_bytes

    console.print("Fetching ZAP versions...")
    zap_versions = await fetch_zap_versions()

    addon = None
    for a in zap_versions.addons:
        if a.id == addon_id:
            addon = a
            break

    if not addon:
        console.print(f"[red]Addon '{addon_id}' not found[/red]")
        raise typer.Exit(1)

    console.print(f"\n[bold]=== {addon.name} ===[/bold]")
    console.print(f"ID: {addon.id}")
    console.print(f"Version: {addon.version}")
    console.print(f"Status: {addon.status}")
    console.print(f"Author: {addon.author}")
    console.print(f"Description: {addon.description}")
    console.print(f"File: {addon.file}")
    console.print(f"Size: {format_bytes(addon.size)}")
    console.print(f"Date: {addon.date}")
    console.print(f"URL: {addon.url}")
    console.print(f"Hash: {addon.hash}")
    console.print(f"Min ZAP Version: {addon.not_before_version}")

    if addon.dependencies:
        console.print("\n[bold]Dependencies:[/bold]")
        for dep in addon.dependencies:
            console.print(f"  - {dep.id} ({dep.version})")
